### PR TITLE
feat: remove strings dependency, optimizations

### PR DESCRIPTION
**Pull request by nilslice**
Pulling in #16, just easier to start from a clean branch that has been rebased.
Thanks @syke99 for leading the effort to get this done!

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -3,12 +3,12 @@ example:
 	tinygo build -o example/tiny_countvowels.wasm -target wasi ./example/countvowels
 	tinygo build -o example/tiny_http.wasm        -target wasi ./example/http
 
-	GOOS=wasip1 GOARCH=wasm go build -o example/std_countvowels.wasm ./example/countvowels
-	GOOS=wasip1 GOARCH=wasm go build -o example/std_http.wasm        ./example/http
+	GOOS=wasip1 GOARCH=wasm go build -tags std -o example/std_countvowels.wasm ./example/countvowels
+	GOOS=wasip1 GOARCH=wasm go build -tags std -o example/std_http.wasm        ./example/http
```
Comment (nilslice) [inline Makefile:7]: added build tags to the files in `example/` so we can keep these single-path locations for the builds and conditionally compile the right code based on which compiler is used. 

the downside is we need to maintain two versions of the examples.

File: Makefile [hunk 2]
```diff
@@ -3,12 +3,12 @@ example:
...
 
 test:
 	extism call example/tiny_countvowels.wasm count_vowels --wasi --input "this is a test" --set-config '{"thing": "1234"}'
 	extism call example/tiny_http.wasm        http_get     --wasi --log-level info --allow-host "jsonplaceholder.typicode.com"
 
-	extism call example/std_countvowels.wasm count_vowels --wasi --input "this is a test" --set-config '{"thing": "1234"}'
-	extism call example/std_http.wasm        http_get     --wasi --log-level info --allow-host "jsonplaceholder.typicode.com"
\ No newline at end of file
+	extism call example/std_countvowels.wasm _start     --wasi --input "this is a test" --set-config '{"thing": "1234"}'
+	extism call example/std_http.wasm        _start     --wasi --log-level info --allow-host "jsonplaceholder.typicode.com"
```
Comment (nilslice) [inline Makefile:14]: all `go build`-based wasm needs to be invoked via `_start`, as it is the only export present when using `wasip1` GOOS.

**Comment by egonelbre** [comment]
I should mention that in [issue ](https://github.com/extism/go-pdk/issues/14) wasn't that the "strings" package was included per se... the main issue is that it's not possible to use the main package without having unicode tables included. i.e. if `go list -deps` includes unicode, then it's still a problem.

The only way to easily solve this is to move http things to a sub-package like `github.com/extism/go-pdk/pdkhttp` -- this would still allow easily using custom method names.

The other approach is to write a custom json encoder that doesn't rely on unicode tables.

**Review by bhelx — APPROVED** [review]
Looks good to me. Are there any consequences to this?

File: Makefile (hunk 1 above)
Comment (nilslice) [reply Makefile:7]: @bhelx ^ this is one consequence but not a breaking change. 

the library does have breaking changes but nothing really extreme

**Comment by nilslice** [comment]
@egonelbre - that's a good suggestion. will consider it!

**Comment by bhelx** [comment]
Anything else we need to merge this?

**Comment by nilslice** [comment]
I think it's good to go!

**Pull request merged**